modules/key-distributor/README.md

```diff
@@ -0,0 +1,166 @@
+send it off to a friend, etc.
+
+This solution allows operators to generate a Google Cloud service account key for another system (e.g. CICD pipeline) or user without
+ever needing to see the key itself, adhering to the principle of least privilege. This is possible using asymmetric encryption
+and the GPG software utility. This assumes the receiving service has a GPG private key which can be used to decrypt the service
+account key, and the deployer of this function has access to the GPG public key.
+
+## Architecture
+
+![](./arch.png)
+
+In this diagram, you can see the Cloud Function is used to broker the Service Account key generation and public key encryption. The operator then only needs to be granted
+access to call the function, but not to create the service account key. The function returns the encrypted key, which the operator can then distribute to
+the application or user that owns the private key.
+
+## Deployment
+```
+module "distributor" {
+  source  = "terraform-google-modules/service-accounts/google//key-distributor"
+
+  project_id      = "<project-id>"
+  org_id          = "<org-id>"
+  public_key_file = "/path/to/pubkey.asc"
+
+  # These folks will be granted access to use the function
+  cfn_members = [
+      "user:someuser@example.com",
+      "group:somegroup@example.com",
+  ]
+}
+```
+
+### Prerequisites
+#### Required APIs:
+
+* `iam.googleapis.com`
+* `cloudbuild.googleapis.com`
+* `cloudfunctions.googleapis.com`
+
+#### Required software:
+
+* [gpg](https://gnupg.org/) - Used for fetching and exporting public keys as well as decrypting
+* [terraform](https://www.terraform.io/downloads.html) - Used for deployment of the Cloud Function
+* [jq](https://stedolan.github.io/jq/) - Used to parse JSON response
+
+On top of the typical terraform setup, we need to export the public key the Cloud Function will use. Make sure you have the key you need in your local
+gpg key chain. If it's in a public key server you can download it with:
+
+```
+gpg --receive-keys <key-id>
+```
+
+Once it's there locally, you can export the ASCII armored version of it with this and store it into a file:
+
+```
+gpg --export --armor <key-id> > pubkey.asc
+```
+
+### Apply
+
+With your variables and public key setup, simply run:
+
+```
+terraform init
+terraform apply
+```
+
+## Usage
+
+When you run the terraform, a new script will appear at `get-key` from a template. This is a basic wrapper
+script that writes an encrypted file after invoking the cloud function. Feel free to take this and expand on it for
+your own use cases, or just run it as is.
+
+```
+./get-key <sa-email>
+Success! Wrote encrypted key to <sa-email>.json.gpg
+```
+
+Now you can safely distribute this gpg-encrypted file to the server or container that has the private key associated
+with it, all while you never had to see the service account key or the private key!
+
+## Validation
+
+Now let's try an end to end demo since we have everything deployed.
```

Comment:
This might be better content to move to the example README?

Reply:
I moved the GPG key export piece to the example and linked to the module README, but I think the main usage instruction should still be the main module README in case it get's lost. Mainly because that README gets populated into the Terraform registry but the example directory would not